### ISSUE TITLE
Update HtmlReport with collapsible sub-steps

### ISSUE THREAD

**Issue by lemonlion** [LightBDD/LightBDD#282]
#### Description
The LightBdd report that is created acts as a specifications document for the service.  However, balancing the level of detail in each test can be difficult.  Often you want fewer steps on the top level that consist of various 'sub' steps, that are sometimes repetitious, or can create noise that distracts from intent.  The developer is able to easily navigate through to those 'sub' steps, but the person reading the report cannot.  I want a way of maintaining that same level of clarity, while still allowing the report reader to view a higher level of granularity. 

I envision this feature working via an attribute "SubStep".  The primary partial class will look identical, but the 'code-behind' class will have some methods that are marked as substeps.  The report writer will pick up on this during execution.

```
public partial class Some_Feature
{
    [Scenario]
    public async Task Health_Check_Endpoint_Is_Successful()
    {
        await Runner.RunScenarioAsync(
            given => Some_step(),
            when => Some_other_step(),
            then => Some_final_step());
    }
}
```

```
public partial class Some_Feature : BaseFixture
{
    private async Task Some_step()
    {
        await Some_substep();
        await Another_substep();
        await A_final_substep();
    }

    private Task Some_other_step()
    {...}

    private Task Some_final_step()
    {...}

    [SubStep]
    private Task Some_substep()
    {...}

    [SubStep]
    private Task Another_substep()
    {...}

    [SubStep]
    private Task A_final_substep()
    {...}
}
```

The output on the report would look like below.  The first step will be expandable on click of the '+' sign.

![lightBddSubstepRetracted](https://github.com/LightBDD/LightBDD/assets/8618373/b67a26e0-fdd5-4d44-a064-ac12ecb3765b)

Likewise they can then retract again by clicking the minus sign.

![lightBddSubstepExpanded](https://github.com/LightBDD/LightBDD/assets/8618373/3b1348a4-6618-4bd4-9fbd-1a04f054c117)

Or even more simplified (without pass/failure for the substeps) would still add great value:

![lightBddSubstepExpandedSimplified](https://github.com/LightBDD/LightBDD/assets/8618373/6af02e9a-efde-445e-9cda-292391a149cd)

I believe this would be a killer feature which would give LightBdd a major edge over the other frameworks.

**Implementation Notes**
I imagine this wouldn't be trivial to implement, as you're probably getting your step names from the expressions, the substeps would be far harder. Perhaps there is some info you could extract from the call stack trace?

#### Progress

- [ ] Feature is implemented,
- [ ] Ensured backward-compatibility,
- [ ] Ensured good debugging experience

**Comment by Suremaker** [comment]
Hi @lemonlion ,

Such feature already exists and is called [composite steps](https://github.com/LightBDD/LightBDD/wiki/Composite-Steps-Definition). Here is the example on how they are rendered in the [html report](https://htmlpreview.github.io/?https://github.com/LightBDD/LightBDD/blob/master/examples/ExampleReports/FeaturesReport.html#feature6).

There is also a page describing more advanced scenarios of data passing between the composites (including DI injections): [sharing-state-with-composite-steps](https://github.com/LightBDD/LightBDD/wiki/Scenario-State-Management#sharing-state-with-composite-steps)

The only part that is not implemented is the collapsing / expanding sub-steps on the report, which I am happy to look at as I have been already thinking about making this improvement.

Please let me know what you think about it.

**Comment by lemonlion** [comment]
Ah just tried the composite steps and it works great! I'll leave this open as a reminder to implement the small but key feature (in terms of noise reduction) of the +/- in the report.  I would suggest that the default behaviour when the report is first opened is that the substeps are *not* expanded out, for clarity.

**Comment by Suremaker** [comment]
Released with LightBDD 3.7.0
